### PR TITLE
misc: improve bug report issue template

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

Come from https://github.com/facebook/docusaurus/issues/4853#issuecomment-848724840

Should help reduce unwanted bug reports. Or do we still need to mention about this in docs too?

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

N/A

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug.md [hunk 1]
```diff
@@ -18,7 +18,19 @@ labels: 'bug, needs triage'
...
-(A clear and concise description of what the bug is)
+### Prerequisites
+
+<!--
+IMPORTANT: Please meet all following the preconditions before submitting an issue.
+Put an X between the brackets on each line to confirm you have completed them.
+-->
+
+- [ ] Are you using the latest version of Docusaurus?
+- [ ] Have you tried running the `yarn clear` command?
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/bug.md:29]:
```suggestion
- [ ] I'm using the latest version of Docusaurus.
- [ ] I have tried the `yarn clear` command.
```

File: .github/ISSUE_TEMPLATE/bug.md [hunk 2]
```diff
@@ -18,7 +18,19 @@ labels: 'bug, needs triage'
 
 ## 🐛 Bug Report
 
-(A clear and concise description of what the bug is)
+### Prerequisites
+
+<!--
+IMPORTANT: Please meet all following the preconditions before submitting an issue.
+Put an X between the brackets on each line to confirm you have completed them.
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/bug.md:25]:
```suggestion
IMPORTANT: Please check the following items before creating a issue.
Mark the checkbox by putting an X between the bracket.
This way we know you've done these steps first.
```

**Review by HonkingGoose — COMMENTED** [review]
How do you feel about this wording?

I think it's clearer on what the user needs to do, and is still understandable if they skip over the comment. 😄

**Comment by lex111** [comment]
@HonkingGoose great suggestions, thanks!

File: .github/ISSUE_TEMPLATE/bug.md [hunk 3]
```diff
@@ -63,7 +76,7 @@ If possible, use https://new.docusaurus.io to create a CodeSandbox reproducible
 - Public site url:
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/bug.md:76]:
```suggestion
- Public site URL:
```

While we're fixing things up, URL is an abbreviation, and in English we capitalize abbreviations. 😄

File: .github/ISSUE_TEMPLATE/bug.md [hunk 4]
```diff
@@ -18,7 +18,20 @@ labels: 'bug, needs triage'
...
+### Prerequisites
+
+<!--
+IMPORTANT: Please check the following items before creating a issue.
+Mark the checkbox by putting an X between the bracket.
+This way we know you've done these steps first.
+-->
+
+- [ ] I'm using the latest version of Docusaurus.
+- [ ] I have tried the `yarn clear` command.
```
Comment (slorber) [inline .github/ISSUE_TEMPLATE/bug.md:30]:
```suggestion
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages
```

I think showing the npm command might be useful for non-frontend-dev users that might be confused by seeing a yarn command.

Also think re-generating lockfile / reinstall is a useful thing to try

**Review by slorber — APPROVED** [review]
Thanks, that looks like a good idea

**Comment by slorber** [comment]
I have also noticed a trick used by some other repos for questions:

https://github.com/software-mansion/react-native-reanimated/blob/master/.github/ISSUE_TEMPLATE/config.yml

image

Maybe we should do the same so that questions end up in the discussions instead of issues?

**Comment by HonkingGoose** [comment]
@lex111 and @slorber, here is the documentation from GitHub for the `config.yml` file that you can use to configure the issue template chooser: 

- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser

**Pull request merged**